tla-bmcmt/src/test/scala/at/forsyte/apalache/tla/bmcmt/TestSymbStateRewriterRowRecord.scala

```diff
@@ -0,0 +1,116 @@
+  private val fieldC = str("d").typed()
+
+  test("""[a |-> 22, b |-> FALSE, c |-> "d"]""") { rewriterType: SMTEncoding =>
+    val recordT = parser("{ a: Int, b: Bool, c: Str }")
+    val record = enumFun(str("a"), fieldA, str("b"), fieldB, str("c"), fieldC).as(recordT)
+
+    val state = new SymbState(record, arena, Binding())
+    val rewriter = create(rewriterType)
+    val nextState = rewriter.rewriteUntilDone(state)
+    assert(solverContext.sat())
+    val cell = nextState.asCell
+    assert(recordT == cell.cellType.toTlaType1)
+
+    expectField(rewriter, nextState, cell, "a", fieldA)
+    expectField(rewriter, nextState, cell, "b", fieldB)
+    expectField(rewriter, nextState, cell, "c", fieldC)
+  }
+
+  test("""[a |-> 22, b |-> FALSE, c |-> "d"].a""") { rewriterType: SMTEncoding =>
+    val recordT = parser("{ a: Int, b: Bool, c: Str }")
+    val record = enumFun(str("a"), fieldA, str("b"), fieldB, str("c"), fieldC).as(recordT)
+    val valueOfA = appFun(record, str("a")).as(IntT1())
+
+    val state = new SymbState(valueOfA, arena, Binding())
+    val rewriter = create(rewriterType)
+    val nextState = rewriter.rewriteUntilDone(state)
+    assert(solverContext.sat())
+    val cell = nextState.asCell
+    assert(fieldA.typeTag.asTlaType1() == cell.cellType.toTlaType1)
+
+    val eq = eql(cell.toNameEx, int(22)).as(BoolT1())
+    assertTlaExAndRestore(rewriter, nextState.setRex(eq))
+  }
+
+  test("""[[a |-> 22, b |-> FALSE, c |-> "d"] EXCEPT !.a = 10]""") { rewriterType: SMTEncoding =>
+    val recordT = parser("{ a: Int, b: Bool, c: Str }")
+    val record = enumFun(str("a"), fieldA, str("b"), fieldB, str("c"), fieldC).as(recordT)
+    val newRecord = except(record, tuple(str("a")).as(TupT1(StrT1())), int(10)).as(recordT)
+
+    val state = new SymbState(newRecord, arena, Binding())
+    val rewriter = create(rewriterType)
+    val nextState = rewriter.rewriteUntilDone(state)
+    assert(solverContext.sat())
+    val cell = nextState.asCell
+    assert(recordT == cell.cellType.toTlaType1)
+
+    expectField(rewriter, nextState, cell, "a", int(10).typed())
+    expectField(rewriter, nextState, cell, "b", fieldB)
+    expectField(rewriter, nextState, cell, "c", fieldC)
+  }
+
+  test("""DOMAIN [a |-> 1, b |-> FALSE, c |-> "d"]""") { rewriterType: SMTEncoding =>
+    val recordT = parser("{ a: Int, b: Bool, c: Str }")
+    val record = enumFun(str("a"), fieldA, str("b"), fieldB, str("c"), fieldC).as(recordT)
+    val domain = dom(record).as(SetT1(StrT1()))
+
+    val state = new SymbState(domain, arena, Binding())
+    val rewriter = create(rewriterType)
+    val nextState = rewriter.rewriteUntilDone(state)
+    assert(solverContext.sat())
+    val cell = nextState.asCell
+
+    val eq = eql(cell.toNameEx, enumSet(str("a"), str("b"), str("c")).as(SetT1(StrT1()))).as(BoolT1())
+    assertTlaExAndRestore(rewriter, nextState.setRex(eq))
+  }
+
+  test("record equality") { rewriterType: SMTEncoding =>
```

Comment:
Can you also add a test for inequality? Specifically `rec.as({a: Int}) /= rec.as({a: Int, b: Int})`

Reply:
This would not pass the type checker. What's the point?

Reply:
Fine, then a test for 2 rows of the same variant. The point is to have both positive tests _and_ negative tests for evaluating `=` on records.